plugins/generator-1.15.2/forge-1.15.2/templates/fluid.java.ftl

```diff
@@ -108,6 +113,18 @@ import net.minecraft.block.material.Material;
 			}
 			</#if>
 
+			<#if data.lightOpacity == 0>
+			@Override
+			public boolean propagatesSkylightDown(BlockState state, IBlockReader reader, BlockPos pos) {
+				return true;
+			}
+			<#elseif data.lightOpacity != 1>
+			@Override
+			public int getOpacity(BlockState state, IBlockReader worldIn, BlockPos pos) {

```

Comment:
I was not aware of this one, lightopacity for block mod element could be improved. Would you mind opening PR adding this functionality to blocks, that currently do it like this: https://github.com/MCreator/MCreator/blob/13e9da386ebfeebbd08d0e89144419ac51474263/plugins/generator-1.15.2/forge-1.15.2/templates/block.java.ftl#L280 (1.15.2 as example, similar with 1.16.5)

Reply:
Is it fine if I disable it for other block bases (except leaves)? I'm not sure how to make it work with stairs/slabs, and I don't think anyone would want to make an opaque fence gate

Reply:
Is it not possible to override it for these blocks too? Or does it not work? Either way, if it is not doable then it is fine to disable them

Reply:
@AlsoSomeoneElse any updates on this one?

Reply:
Got it to work with other block bases, but it can look weird with some (stairs, slabs). I'm thinking of adding an element converter to change the light opacity values to 0-15 (like the one for block luminance), with -1 as a "default value" for some block bases (otherwise stairs with the default 255 light opacity would convert to 15, which makes them fully opaque)

Reply:
I think you can hardcode light opacity for these block bases in the ftl templates by checking if block base is one of them, so you avoid doing the whole converter thing and keep changes isolated only to templates

Reply:
What about the opacity range? Should it remain 0-255?

Reply:
I think so. If you disable the field in UI for block bases using "default value", you don't need to save default value as it will be used as default inside the template that should detect block base is used

Reply:
The field wouldn't be disabled for other block bases, the -1 is there because all the other values would make blocks behave differently than they currently do. As for the range, I think it'd be better to change the max value to 15, there's no point in setting an opacity higher than that

Reply:
Ok, so if the new range is between 0-15 then yes, you should scale the value and make a converter. In the past, it was between 0-255.

Regarding -1, if you don't even get that value for block bases that should use vanilla value, you don't need to store that value. Just set to the right value inside the template, eg. (pseudocode):

```

IF base = stairs

returnopacity 0

ELSEIF  base = some other base

returnopacity 4

ELSE

returnopacity ${data.opacity}

```